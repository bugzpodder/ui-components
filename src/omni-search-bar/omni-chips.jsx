// @flow
import Avatar from "@material-ui/core/Avatar";
import Chip from "@material-ui/core/Chip";
import React, { Fragment } from "react";
import SearchIcon from "@material-ui/icons/Search";
import { OMNI_KEY } from "@grail/lib";

type Props = {
  searchOptions: SearchOptionsV2,
  setSearchOptions?: SearchOptionsV2 => any,
};

export const OmniChips = (props: Props) => {
  const { searchOptions, setSearchOptions } = props;
  const onDelete = setSearchOptions
    ? deletedName => {
      const newSearchOptions = searchOptions.filter(({ name }) => deletedName !== name);
      setSearchOptions(newSearchOptions);
    }
    : undefined;
  return (
    <Fragment>
      {searchOptions.map(searchOption => {
        const { name, values = [] } = searchOption;
        const displayName = name === OMNI_KEY ? "" : `${name}: `;
        const initials = displayName
          .split(" ")
          .map(word => word.substring(0, 1).toUpperCase())
          .join("");
        return (
          <Chip
            key={name}
            data-testid={name}
            className="margin-right-10"
            color="primary"
            variant="outlined"
            aria-label={name}
            label={`${displayName}${values.join(", ")}`}
            avatar={<Avatar>{initials || <SearchIcon />}</Avatar>}
            onDelete={onDelete && onDelete.bind(this, name)}
          />
        );
      })}
    </Fragment>
  );
};
