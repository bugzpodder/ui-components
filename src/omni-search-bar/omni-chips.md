### Example

```js
const ExampleBlock = require("../test-utils").ExampleBlock;
const ExampleWrapper = require("../test-utils").ExampleWrapper;

const {
  OMNI_TEXT_SEARCH_TYPE,
  DATETIME_SEARCH_TYPE,
  FULL_TEXT_SEARCH_TYPE,
  LIKE_TEXT_SEARCH_TYPE,
  OMNI_KEY,
  SET_OMNI_FIELD_COMMAND,
} = require("@grail/lib");
const { OmniChips } = require("./index");

const initialSearchOptions = [
  {
    name: OMNI_KEY,
    type: OMNI_TEXT_SEARCH_TYPE,
    searchFields: ["label", "samples.label", "samples.previousLabel"],
    values: ["omni value"],
  },
  {
    name: "Labels",
    type: LIKE_TEXT_SEARCH_TYPE,
    searchFields: ["label"],
    values: ["labels value"],
  },
  {
    name: "Operation Time",
    type: DATETIME_SEARCH_TYPE,
    searchFields: ["operationTime"],
    values: ["2018-04-20", "2019-04-20"],
  },
  {
    name: "Status",
    type: FULL_TEXT_SEARCH_TYPE,
    searchFields: ["status"],
    values: ["status value", "Another Value"],
  },
];

class OmniChipsContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = { searchOptions: initialSearchOptions };
    this.addOmniSearchCommand = this.addOmniSearchCommand.bind(this);
  }
  addOmniSearchCommand({ command, omniFieldName, omniValues }) {
    console.log(command, omniFieldName, omniValues);
    if (command === SET_OMNI_FIELD_COMMAND) {
      this.setState(({ searchOptions }) => {
        searchOptions = searchOptions
          .map(searchOption => {
            let { name, values } = searchOption;
            if (omniFieldName === name) {
              values = omniValues;
            }
            return { ...searchOption, values };
          })
          .filter(({ values }) => values && values.length);
        return { searchOptions };
      });
    }
  }
  render() {
    return (
      <div>
        <OmniChips searchOptions={this.state.searchOptions} addOmniSearchCommand={this.addOmniSearchCommand} />
        <ExampleBlock strongHeader="State" content={this.state} />
      </div>
    );
  }
}

<ExampleWrapper>
  <OmniChipsContainer />
</ExampleWrapper>;
```
