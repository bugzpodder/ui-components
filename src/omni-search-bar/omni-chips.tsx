import Chip from "@material-ui/core/Chip";

import React from "react";
import styles from "./omni.module.scss";
import {
  OMNI_KEY,
  OmniSearchCommand,
  SET_OMNI_FIELD_COMMAND,
  SearchOptionV2,
} from "@grailbio/lib";

type Props = {
  /** Search Options. */
  searchOptions: SearchOptionV2[];
  /** Function to set searchOptions */
  addOmniSearchCommand?: (x0: OmniSearchCommand) => any;
};

export const OmniChips: React.FC<Props> = props => {
  const { searchOptions, addOmniSearchCommand } = props;
  return (
    <div className={styles.omniChips} data-testid="omni-chips-container">
      {searchOptions.map(searchOption => {
        const { name, values = [] } = searchOption;
        const displayName = name === OMNI_KEY ? "" : `${name}: `;
        const onDelete = addOmniSearchCommand
          ? (omniFieldName, valueIndex) => {
              const omniValues = [...values];
              omniValues.splice(valueIndex, 1);
              addOmniSearchCommand({
                command: SET_OMNI_FIELD_COMMAND,
                omniFieldName,
                omniValues,
              });
            }
          : undefined;
        return values.map((value, valueIndex) => (
          <Chip
            key={`${name}-${valueIndex}`}
            data-testid={`${name}-${valueIndex}`}
            classes={{ root: styles.chip }}
            color="primary"
            variant="default"
            aria-label={name}
            label={`${displayName}${value}`}
            onDelete={onDelete && onDelete.bind(this, name, valueIndex)}
          />
        ));
      })}
    </div>
  );
};
