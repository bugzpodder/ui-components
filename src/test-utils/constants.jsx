// @flow
import React from "react";
import { EDC, PIPELINE } from "@grail/lib";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const GRAIL = "https://grail.com/wp-content/uploads/2016/12/cropped-final-favicon-32x32.png";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 1000; i++) {
  // Create about 17,000 records.
  LARGE_EXAMPLE_TABLE_DATA.push(...EXAMPLE_TABLE_DATA);
}

export const EXTRA_LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 100; i++) {
  // Create about 1,700,000 records
  EXTRA_LARGE_EXAMPLE_TABLE_DATA.push(...LARGE_EXAMPLE_TABLE_DATA);
}

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];

export const ALPHABET_ANIMALS_DATA: Array<Array<?string>> = [
  ["alligator", "bat", "cat", "dog", "elephant", "fox"],
  ["goat", "horse", "iguana", "jaguar", "kangaroo", "lion"],
  ["monkey", "narwhal", "octopus", "penguin", "quail", "raccoon"],
  ["seal", "tiger", "umbrella bird", "vulture", "whale", "x-ray tetra"],
  ["yak", "zebra", null, null, null, null],
];

export const TEST_EXTERNAL_DOMAINS: Map<string, string> = new Map().set(EDC, "edc.com").set(PIPELINE, "pipeline.com");
