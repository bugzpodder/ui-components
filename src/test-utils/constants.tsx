import React from "react";

export const COUNTRY_NAMES = [
  "Afghanistan",
  "Akrotiri",
  "Albania",
  "Algeria",
  "American Samoa",
  "Andorra",
  "Angola",
  "Anguilla",
  "Antarctica",
  "Antigua and Barbuda",
  "Argentina",
  "Armenia",
  "Aruba",
  "Ashmore and Cartier Islands",
  "Australia",
  "Austria",
  "Azerbaijan",
  "Bahamas, The",
  "Bahrain",
  "Bangladesh",
  "Barbados",
  "Bassas da India",
  "Belarus",
  "Belgium",
  "Belize",
  "Benin",
  "Bermuda",
  "Bhutan",
  "Bolivia",
  "Bosnia and Herzegovina",
  "Botswana",
  "Bouvet Island",
  "Brazil",
  "British Indian Ocean Territory",
  "British Virgin Islands",
  "Brunei",
  "Bulgaria",
  "Burkina Faso",
  "Burma",
  "Burundi",
  "Cambodia",
  "Cameroon",
  "Canada",
  "Cape Verde",
  "Cayman Islands",
  "Central African Republic",
  "Chad",
  "Chile",
  "China",
  "Christmas Island",
  "Clipperton Island",
  "Cocos (Keeling) Islands",
  "Colombia",
  "Comoros",
  "Congo, Democratic Republic of the",
  "Congo, Republic of the",
  "Cook Islands",
  "Coral Sea Islands",
  "Costa Rica",
  "Cote d'Ivoire",
  "Croatia",
  "Cuba",
  "Cyprus",
  "Czech Republic",
  "Denmark",
  "Dhekelia",
  "Djibouti",
  "Dominica",
  "Dominican Republic",
  "Ecuador",
  "Egypt",
  "El Salvador",
  "Equatorial Guinea",
  "Eritrea",
  "Estonia",
  "Ethiopia",
  "Europa Island",
  "Falkland Islands (Islas Malvinas)",
  "Faroe Islands",
  "Fiji",
  "Finland",
  "France",
  "French Guiana",
  "French Polynesia",
  "French Southern and Antarctic Lands",
  "Gabon",
  "Gambia, The",
  "Gaza Strip",
  "Georgia",
  "Germany",
  "Ghana",
  "Gibraltar",
  "Glorioso Islands",
  "Greece",
  "Greenland",
  "Grenada",
  "Guadeloupe",
  "Guam",
  "Guatemala",
  "Guernsey",
  "Guinea",
  "Guinea-Bissau",
  "Guyana",
  "Haiti",
  "Heard Island and McDonald Islands",
  "Holy See (Vatican City)",
  "Honduras",
  "Hong Kong",
  "Hungary",
  "Iceland",
  "India",
  "Indonesia",
  "Iran",
  "Iraq",
  "Ireland",
  "Isle of Man",
  "Israel",
  "Italy",
  "Jamaica",
  "Jan Mayen",
  "Japan",
  "Jersey",
  "Jordan",
  "Juan de Nova Island",
  "Kazakhstan",
  "Kenya",
  "Kiribati",
  "Korea, North",
  "Korea, South",
  "Kuwait",
  "Kyrgyzstan",
  "Laos",
  "Latvia",
  "Lebanon",
  "Lesotho",
  "Liberia",
  "Libya",
  "Liechtenstein",
  "Lithuania",
  "Luxembourg",
  "Macau",
  "Macedonia",
  "Madagascar",
  "Malawi",
  "Malaysia",
  "Maldives",
  "Mali",
  "Malta",
  "Marshall Islands",
  "Martinique",
  "Mauritania",
  "Mauritius",
  "Mayotte",
  "Mexico",
  "Micronesia, Federated States of",
  "Moldova",
  "Monaco",
  "Mongolia",
  "Montserrat",
  "Morocco",
  "Mozambique",
  "Namibia",
  "Nauru",
  "Navassa Island",
  "Nepal",
  "Netherlands",
  "Netherlands Antilles",
  "New Caledonia",
  "New Zealand",
  "Nicaragua",
  "Niger",
  "Nigeria",
  "Niue",
  "Norfolk Island",
  "Northern Mariana Islands",
  "Norway",
  "Oman",
  "Pakistan",
  "Palau",
  "Panama",
  "Papua New Guinea",
  "Paracel Islands",
  "Paraguay",
  "Peru",
  "Philippines",
  "Pitcairn Islands",
  "Poland",
  "Portugal",
  "Puerto Rico",
  "Qatar",
  "Reunion",
  "Romania",
  "Russia",
  "Rwanda",
  "Saint Helena",
  "Saint Kitts and Nevis",
  "Saint Lucia",
  "Saint Pierre and Miquelon",
  "Saint Vincent and the Grenadines",
  "Samoa",
  "San Marino",
  "Sao Tome and Principe",
  "Saudi Arabia",
  "Senegal",
  "Serbia and Montenegro",
  "Seychelles",
  "Sierra Leone",
  "Singapore",
  "Slovakia",
  "Slovenia",
  "Solomon Islands",
  "Somalia",
  "South Africa",
  "South Georgia and the South Sandwich Islands",
  "Spain",
  "Spratly Islands",
  "Sri Lanka",
  "Sudan",
  "Suriname",
  "Svalbard",
  "Swaziland",
  "Sweden",
  "Switzerland",
  "Syria",
  "Taiwan",
  "Tajikistan",
  "Tanzania",
  "Thailand",
  "Timor-Leste",
  "Togo",
  "Tokelau",
  "Tonga",
  "Trinidad and Tobago",
  "Tromelin Island",
  "Tunisia",
  "Turkey",
  "Turkmenistan",
  "Turks and Caicos Islands",
  "Tuvalu",
  "Uganda",
  "Ukraine",
  "United Arab Emirates",
  "United Kingdom",
  "United States #1",
  "Uruguay",
  "Uzbekistan",
  "Vanuatu",
  "Venezuela",
  "Vietnam",
  "Virgin Islands",
  "Wake Island",
  "Wallis and Futuna",
  "West Bank",
  "Western Sahara",
  "Yemen",
  "Zambia",
  "Zimbabwe",
];

export const COUNTRIES: Array<{
  label: string;
  value: string;
}> = COUNTRY_NAMES.map(name => ({ label: name })).map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const GRAIL =
  "https://grail.com/wp-content/uploads/2016/12/cropped-final-favicon-32x32.png";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  {
    word: "Pneumonoultramicroscopicsilicovolcanoconiosis",
    origin: "General English",
  },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 1000; i++) {
  // Create about 17,000 records.
  LARGE_EXAMPLE_TABLE_DATA.push(...EXAMPLE_TABLE_DATA);
}

export const EXTRA_LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 100; i++) {
  // Create about 1,700,000 records
  EXTRA_LARGE_EXAMPLE_TABLE_DATA.push(...LARGE_EXAMPLE_TABLE_DATA);
}

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct
        trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: (
      <div>
        Over 2500 rounds had been fired with the media claiming that only a few
        emus had been killed.
      </div>
    ),
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: (
      <div>
        Sir George Pearce wages war on the local population of emus in
        Australia.
      </div>
    ),
  },
];

export const ALPHABET_ANIMALS_DATA: Array<Array<string | null>> = [
  ["alligator", "bat", "cat", "dog", "elephant", "fox"],
  ["goat", "horse", "iguana", "jaguar", "kangaroo", "lion"],
  ["monkey", "narwhal", "octopus", "penguin", "quail", "raccoon"],
  ["seal", "tiger", "umbrella bird", "vulture", "whale", "x-ray tetra"],
  ["yak", "zebra", null, null, null, null],
];

export const TEST_EXTERNAL_DOMAINS: Map<string, string> = new Map()
  .set("EDC", "edc.com")
  .set("pipeline", "pipeline.com");
