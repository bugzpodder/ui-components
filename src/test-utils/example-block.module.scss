.container {
  margin: 1rem 0;

  .header {
    font-family: monospace;
  }

  .text {
    background: #f6f8fa;
    font-size: 1rem;
    padding: .5rem;
  }
}