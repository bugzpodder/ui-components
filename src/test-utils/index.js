// @flow
export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
