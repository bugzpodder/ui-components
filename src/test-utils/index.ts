export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  COUNTRY_NAMES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
  TEST_EXTERNAL_DOMAINS,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
