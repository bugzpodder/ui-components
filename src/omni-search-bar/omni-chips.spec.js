// @flow
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { LIKE_TEXT_SEARCH_TYPE, OMNI_KEY, SET_OMNI_FIELD_COMMAND } from "@grailbio/lib";
import { OmniChips } from "./omni-chips";
import { TestWrapper } from "../test-utils";
import { cleanup, fireEvent, render } from "@testing-library/react";

afterEach(cleanup);

const searchOptions = [
  {
    name: OMNI_KEY,
    type: LIKE_TEXT_SEARCH_TYPE,
    values: ["test", "Test 2"],
  },
  {
    name: "Search Field",
    type: LIKE_TEXT_SEARCH_TYPE,
    values: ["test", "test 2", "test 3"],
  },
  {
    name: "Empty Search Field",
    type: LIKE_TEXT_SEARCH_TYPE,
    values: undefined,
  },
];

describe("OmniChips", () => {
  it("renders", () => {
    const { container } = render(
      <TestWrapper>
        <OmniChips searchOptions={searchOptions} />
      </TestWrapper>,
    );
    expect(container).toMatchSnapshot();
  });
  it("deletes", () => {
    const mockAddCommand = jest.fn(result => result);
    const { container } = render(
      <TestWrapper>
        <OmniChips
          searchOptions={searchOptions}
          addOmniSearchCommand={mockAddCommand}
        />
      </TestWrapper>,
    );
    expect(container).toMatchSnapshot();
    // TODO(nsawas): Fix this.
    // const deleteButton = container.querySelector("[data-testid='Search Field-1'] .MuiChip-deleteIcon");
    // fireEvent.click(deleteButton);
    // expect(mockAddCommand.mock.results[0].value).toEqual({
    //   command: SET_OMNI_FIELD_COMMAND,
    //   omniFieldName: "Search Field",
    //   omniValues: ["test", "test 3"],
    // });
    // expect(container).toMatchSnapshot();
  });
});
