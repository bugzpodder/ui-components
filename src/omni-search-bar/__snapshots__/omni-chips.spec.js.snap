// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OmniSearchBar deletes 1`] = `
<div>
  <div
    class="omniChips"
    data-testid="omni-chips-container"
  >
    <div
      aria-label="omni"
      class="MuiChip-root chip MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
      data-testid="omni-0"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        test
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <div
      aria-label="omni"
      class="MuiChip-root chip MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
      data-testid="omni-1"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        Test 2
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <div
      aria-label="Search Field"
      class="MuiChip-root chip MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
      data-testid="Search Field-0"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        Search Field: test
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <div
      aria-label="Search Field"
      class="MuiChip-root chip MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
      data-testid="Search Field-1"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        Search Field: test 2
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <div
      aria-label="Search Field"
      class="MuiChip-root chip MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
      data-testid="Search Field-2"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        Search Field: test 3
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`OmniSearchBar deletes 2`] = `
<div>
  <div
    class="omniChips"
    data-testid="omni-chips-container"
  >
    <div
      aria-label="omni"
      class="MuiChip-root chip MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
      data-testid="omni-0"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        test
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <div
      aria-label="omni"
      class="MuiChip-root chip MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
      data-testid="omni-1"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        Test 2
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <div
      aria-label="Search Field"
      class="MuiChip-root chip MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
      data-testid="Search Field-0"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        Search Field: test
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <div
      aria-label="Search Field"
      class="MuiChip-root chip MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
      data-testid="Search Field-1"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        Search Field: test 2
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
    <div
      aria-label="Search Field"
      class="MuiChip-root chip MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
      data-testid="Search Field-2"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        Search Field: test 3
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`OmniSearchBar renders 1`] = `
<div>
  <div
    class="omniChips"
    data-testid="omni-chips-container"
  >
    <div
      aria-label="omni"
      class="MuiChip-root chip MuiChip-colorPrimary"
      data-testid="omni-0"
    >
      <span
        class="MuiChip-label"
      >
        test
      </span>
    </div>
    <div
      aria-label="omni"
      class="MuiChip-root chip MuiChip-colorPrimary"
      data-testid="omni-1"
    >
      <span
        class="MuiChip-label"
      >
        Test 2
      </span>
    </div>
    <div
      aria-label="Search Field"
      class="MuiChip-root chip MuiChip-colorPrimary"
      data-testid="Search Field-0"
    >
      <span
        class="MuiChip-label"
      >
        Search Field: test
      </span>
    </div>
    <div
      aria-label="Search Field"
      class="MuiChip-root chip MuiChip-colorPrimary"
      data-testid="Search Field-1"
    >
      <span
        class="MuiChip-label"
      >
        Search Field: test 2
      </span>
    </div>
    <div
      aria-label="Search Field"
      class="MuiChip-root chip MuiChip-colorPrimary"
      data-testid="Search Field-2"
    >
      <span
        class="MuiChip-label"
      >
        Search Field: test 3
      </span>
    </div>
  </div>
</div>
`;
